include/highfive/h5easy_bits/H5Easy_misc.hpp

```diff
@@ -59,16 +48,146 @@ inline void createGroupsToDataSet(File& file, const std::string& path) {
     }
 }
 
-}  // namespace detail
+// Generate error-stream and return "Exception" (not yet thrown).
+inline Exception error(const File& file,
+                       const std::string& path,
+                       const std::string& message) {
+    std::ostringstream ss;
+    ss << message << std::endl
+       << "Path: " << path << std::endl
+       << "Filename: " << file.getName() << std::endl;
+    return Exception(ss.str());
+}
+
+// Generate specific dump error
+inline Exception dump_error(File& file, const std::string& path)
+{
+    if (file.getObjectType(path) == ObjectType::Dataset) {
+        return error(file, path,
+            "H5Easy: Dataset already exists, dump with H5Easy::DumpMode::Overwrite "
+            "to overwrite (with an array of the same shape).");
+    } else {
+        return error(file, path,
+            "H5Easy: path exists, but does not correspond to a Dataset. Dump not possible.");
+    }
+}
 
-inline size_t getSize(const File& file, const std::string& path) {
-    return file.getDataSet(path).getElementCount();
+// get a opened DataSet: nd-array
+template <class T>
+inline DataSet initDataset(File& file,
+                           const std::string& path,
```

Comment:
At a glance I see four implementations that share similar concepts. Not a blocker, I'm more in favour of unified solutions if that could be applied here.

Reply:
I'm not against such a clean up. One would have to see if it becomes something readable/maintainable. Maybe you could open a PR when this is PR is merged?